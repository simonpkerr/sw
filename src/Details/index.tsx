import { useEffect, useState } from 'react';
import { findFilms, findStarships } from '../api';

type DetailsProps = { films: null | string[]; starships: null | string[] };
const Details = ({ location }: any) => {
  const [characterDetails, setCharacterDetails] = useState<DetailsProps>({
    films: null,
    starships: null,
  });
  useEffect(() => {
    const getDetails = async () => {
      const films = location?.state?.films;
      const starships = location?.state?.starships;

      if (films && starships) {
        const filmData = await findFilms(films);
        const starshipData = await findStarships(starships);
        setCharacterDetails({ films: filmData, starships: starshipData });
      }
    };

    getDetails();
  }, [location]);

  const character = location?.state;

  return character ? (
    <>
      <h1>{character.name}</h1>
      <ul>
        <li>Hair colour: {character.hair_color}</li>
        <li>Eye colour: {character.eye_color}</li>
        <li>Gender: {character.gender}</li>
        <li>Home planet: {character.homeworld}</li>
      </ul>
      {characterDetails.films && characterDetails.films.length > 0 && (
        <>
          <h2>Films</h2>
          <ul>
            {characterDetails.films.map((film, i) => (
              <li key={i}>{film}</li>
            ))}
          </ul>
        </>
      )}
      {characterDetails.starships && characterDetails.starships.length > 0 && (
        <>
          <h2>Starships</h2>
          <ul>
            {characterDetails.starships.map((starship, i) => (
              <li key={i}>{starship}</li>
            ))}
          </ul>
        </>
      )}
    </>
  ) : null;
};

export default Details;
