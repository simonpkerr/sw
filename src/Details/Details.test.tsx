import Details from '.';
describe('Details', () => {
  it('should show the character details passed from the list', () => {});

  it('should show a favourites button', () => {});

  it('should load in films and starships', () => {});
});
