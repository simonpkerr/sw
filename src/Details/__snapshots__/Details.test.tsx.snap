// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should show the character details, films and starships passed from the list 1`] = `
<div>
  <h1>
    person
  </h1>
  <ul>
    <li>
      Hair colour: 
      blue
    </li>
    <li>
      Eye colour: 
      red
    </li>
    <li>
      Gender: 
      m
    </li>
    <li>
      Home planet: 
      tatooine
    </li>
  </ul>
  <h2>
    Films
  </h2>
  <ul>
    <li>
      film title 1
    </li>
  </ul>
  <h2>
    Starships
  </h2>
  <ul>
    <li>
      starship 1
    </li>
  </ul>
  <button>
    Add to favourites
  </button>
</div>
`;
